### PR TITLE
Convert all logging to Serilog

### PR DESCRIPTION
**Pull request by dkbennett**
## Summary of the pull request

Log conversion from serilog, very similar changes as: https://github.com/microsoft/DevHomeAzureExtension/pull/143

## Detailed description of the pull request / Additional comments
* DevHome.Logging removed.
* Serilog and related nuget packages added.
* appsettings.json added to the project with log settings.
* All static logging converted over to Serilog
* Some log levels were changed for spammier components.
* New logging added in places using lazy initialization.
* TestContext sink added in place of TestListener, which writes Serilog events to the TestContext.
* Updated all test setups to automatically log and output log messages to TestContext. This means any logs generated during test runs by components will be captured in both a temporary log and in the test output for easy analysis.
* Updated testconsole with logging setup so ad hoc testing with it is already configured for logging.

Log formatting changes:

* Log format was changed to be more compact, easier to parse and be human-readable.
* Log format is similar to Serilog's default modified to be close to the previous log format.
* Single log instead of multiple logs.
Log has a daily interval where it will be a new file, rolling over 30 days.

## Validation steps performed

* Ran all tests and ensured they are producing logs and TestContext messages as well as passing.
* Scenario tested widgets and verified logging.
* Cloned a github repository by url

## PR checklist
- [ ] Closes #xxx
- [x] Tests added/passed
- [x] Documentation updated

### CODE REVIEW

File: src/GitHubExtensionServer/Program.cs [hunk 1]
```diff
@@ -140,6 +152,7 @@ private static void LogPackageInformation()
               "MicrosoftWindows.Client.WebExperience_cw5n1h2txyewy",
               "Microsoft.Windows.DevHome_8wekyb3d8bbwe",
               "Microsoft.Windows.DevHomeGitHubExtension_8wekyb3d8bbwe",
+              "Microsoft.Windows.DevHomeGitHubExtension.Dev_8wekyb3d8bbwe",
```
Comment (krschau) [inline src/GitHubExtensionServer/Program.cs:155]: Should this not be included, or should we include canary as well?

File: src/GitHubExtension/Widgets/GitHubUserWidget.cs [hunk 2]
```diff
@@ -309,7 +307,7 @@ public override string GetData(WidgetPageState page)
     {
         return page switch
         {
-            WidgetPageState.SignIn => new JsonObject { { "message", Resources.GetResource(@"Widget_Template/SignInRequired", Log.Logger()) } }.ToJsonString(),
+            WidgetPageState.SignIn => new JsonObject { { "message", Resources.GetResource(@"Widget_Template/SignInRequired", Log) } }.ToJsonString(),
```
[the remark this answers is not in the document]
Comment (dkbennett) [reply src/GitHubExtension/Widgets/GitHubUserWidget.cs:310]: It's passing the log into the GetResource method, which will log errors with resource loading. The log isn't part of the JSON, only the string that comes back from the method.

File: src/GitHubExtensionServer/Program.cs (hunk 1 above)
Comment (dkbennett) [reply src/GitHubExtensionServer/Program.cs:155]: We should include all of the devhome and extension names so the logging of packages present is complete.

**Review by krschau — APPROVED** [review]
Not sure if you wanted to include the other flavor/extensions from my other comment, but can be done after.

**Pull request merged**